### ISSUE TITLE
Unable to verify jwt against an RSA PUBLIC KEY

### ISSUE THREAD

**Issue by Ironirc** [noha/JSONWebToken#12]
For this I had to extend LcRSA
```
LcRSA>>fromRsaPublicKeyPemString: pemString
	| nullp bio |
	nullp := FFIExternalObject new.
	bio := LcMemoryBIO fromString: (self cr2lf: pemString).
	[ handle := LcLibCrypto current
		apiPemReadBioRsaPublicKey: bio handle
		with: nullp
		with: nullp
		with: nullp.
	self errorIfNull: handle ]
		ensure: [ bio close ]
```
```
LcRSA class>>fromRsaPublicKeyPemString: pemString
	^ self new
		fromRsaPublicKeyPemString: pemString;
		yourself
```

Extend LcLibCrypto
```
LcLibCrypto>>apiPemReadBioRsaPublicKey: bio with: rsa with: cb with: uv
	^ self
		ffiCall:
			#(RSA * PEM_read_bio_RSAPublicKey #(BIO * bio , RSA ** rsa , FFIExternalObject cb , FFIExternalObject uv))
		module: self library
```

Then override:
```
JWARSASHA256 class>>checkSignature: parts withKey: key
...
rsa := (key includesSubstring: 'BEGIN RSA PUBLIC KEY')
		ifTrue: [ LcRSA fromRsaPublicKeyPemString: key ]
		ifFalse: [ LcRSA fromPemString: key ].
	pubKey := LcEvpPublicKey setRSA: rsa.
...
```

On top of this I also have a non rsa variant (BEGIN PUBLIC KEY), which I don't use/need at this moment. If you want I can provide you with the LcLibCrypto method and LcRSA method.

**Comment by noha** [comment]
The LcRSA and LcLibCrypto are not part of this repository but of https://github.com/PierceNg/OpenSSL-Pharo. Can you add a pull request there and then we add the checkSignature here?

**Comment by Ironirc** [comment]
Is it OK to include the extensions of LcRSA and LcLibCrypto in package JSONWebToken-OpenSSL?
Or should I first do a pull request on https://github.com/PierceNg/OpenSSL-Pharo ?

**Comment by noha** [comment]
The general rule would be that if it is convenient or helper method of JWT to OPenSSL it should be an extension method. But these methods look useful to OpenSSL and that's why I think they should be there.

**Comment by Ironirc** [comment]
So guess we'll have to wait for this pull request to be handled?
https://github.com/PierceNg/OpenSSL-Pharo/pull/4

**Comment by noha** [comment]
Yes. I understand that you want to proceed. But if we hack these things away the software dies. So it is an attempt to make it proper. We can still go back having the extrension methods if PierceNg is not responding (which he will soon). If you want to proceed it sometimes better to fork those projects and apply your changeds there and open pull requests. So you can proceed and move back to other repos when the PRs have been integrated

**Comment by noha** [comment]
solved via #13

**Comment by Ironirc** [comment]
Thanks Norbert!
